core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -641,14 +643,28 @@ public UnassignedIterator iterator() {
          * Should be used with caution, typically,
          * the correct usage is to removeAndIgnore from the iterator.
          * @see #ignored()
-         * @see UnassignedIterator#removeAndIgnore()
+         * @see UnassignedIterator#removeAndIgnore(Optional)
          * @see #isIgnoredEmpty()
+         * @return true iff the decision caused a change to the unassigned info
          */
-        public void ignoreShard(ShardRouting shard) {
+        public boolean ignoreShard(ShardRouting shard, Optional<Decision> decision) {
```

Comment:
with the suggestion I made, we don't need to pass an optional now, we aways have a decision.